### ISSUE TITLE
React setState did not rerender the page

### ISSUE THREAD

**Issue by Anthonykung** [reactjs/express-react-views#165]
Hi, I'm having trouble getting the page rerendered with updated data. I have an array of arrays and I would like to render just 1 array at a time with some button to change the state to the next array. But I don't seem to be able to do this with the view engine for some reason, is it not supported?

I have made this simple component to test state change that increments an index
```jsx
import React from 'react';

export class Test extends React.Component {
  constructor(props) {
    super(props);
    this.state = { index: 0 };
    this.increment = this.increment.bind(this);
  }

  increment() {
    let idx = this.state.index + 1;
    if (idx > 10) {
      idx = 10;
    }
    this.setState({ index: idx});
  }

  render() {
    return (
      <div>
        <h1>
          Index: {this.state.index}
        </h1>
        <br />
        <button onClick={this.increment}>
          Increment
        </button>
      </div>
    );
  }
}
```

And this function that returns JSX
```jsx
const React = require('react');
import { Body } from './default';
import { Test } from './test';

export default function TestFunction(props) {
  return (
    <Body>
      <Test />
    </Body>
  );
}
```

It works on React editors but when rendering with the express view engine it doesn't work at all. I'm limited to express due to the nature of my project.

**Comment by Ming-desu** [comment]
I think the reason why it is not re-rendering your component based on their state is that express-react-views are meant only for server-side rendering,

**Comment by Anthonykung** [comment]
Oh I see, so the only way that I can do something like this is to send a POST request and change it right?

**Comment by Ming-desu** [comment]
Yep, that is my current solution as well. I think it is impossible for a server side rendering to have a reactive state. Since the re-rendering process only occurred in client side with React itself that is why I think it is impossible to have an automatic re-rendering when the state changes, unless you will trigger it manually via JavaScript.

**Comment by BrinePlayz** [comment]
> Yep, that is my current solution as well. I think it is impossible for a server side rendering to have a reactive state. Since the re-rendering process only occurred in client side with React itself that is why I think it is impossible to have an automatic re-rendering when the state changes, unless you will trigger it manually via JavaScript.

😭
